tiledb/ml/readers/pytorch.py

```diff
@@ -87,17 +94,45 @@ def __iter__(self) -> Iterator[DataType]:
             iter_start = worker_id * per_worker
             iter_end = min(iter_start + per_worker, rows)
 
+        offsets = range(iter_start, iter_end, self.batch_size)
+
+        # Shuffle offsets in case we need batch shuffling
+        if self.batch_shuffle:
+            random.seed()
```

Comment:
`random.seed` is typically called once at the beginning, e.g. explicitly by the library user or an `tiledb.ml.init()` function to initialize the library. Alternatively one can use a [Random](https://docs.python.org/3/library/random.html#random.Random) instance that encapsulates the seed and pass it to `__init__`.